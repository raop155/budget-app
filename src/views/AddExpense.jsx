import React from 'react';
import { addExpense } from '../actions/expenses';
import { connect } from 'react-redux';
import ExpenseForm from '../components/ExpenseForm';

const AddExpense = props => {
  return (
    <div>
      <h1>Add Expense</h1>
      <ExpenseForm
        onSubmit={expense => {
          props.dispatch(addExpense(expense));
          props.history.push('/');
        }}
      />
    </div>
  );
};

export default connect()(AddExpense);
