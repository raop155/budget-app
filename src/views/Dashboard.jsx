import React from 'react';
import ExpenseListFilters from '../components/ExpenseListFilters';
import ExpenseList from '../components/ExpenseList';

export const Dashboard = () => {
  return (
    <div>
      <ExpenseListFilters />
      <ExpenseList />
    </div>
  );
};

export default Dashboard;
