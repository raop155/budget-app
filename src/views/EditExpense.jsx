import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from '../components/ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export const EditExpense = (props) => {
  const onSubmit = (expense) => {
    props.startEditExpense({ id: props.expense.id }, expense);
    props.history.push('/');
  };

  const handleRemoveExpense = () => {
    props.startRemoveExpense({ id: props.expense.id });
    props.history.push('/');
  };

  return (
    <div>
      <h1>Edit Expense</h1>
      <ExpenseForm expense={props.expense} onSubmit={onSubmit} />
      <button onClick={handleRemoveExpense}>Remove</button>
    </div>
  );
};

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id),
  };
};

const mapDispatchToProps = {
  startEditExpense: ({ id }, expense) => startEditExpense({ id }, expense),
  startRemoveExpense: ({ id }) => startRemoveExpense({ id }),
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpense);
