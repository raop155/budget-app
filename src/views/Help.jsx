import React from 'react';

const Help = () => {
  return <div>Help View</div>;
};

export default Help;
