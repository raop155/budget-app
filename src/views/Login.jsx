import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const Login = ({ startLogin }) => {
  return (
    <>
      <h1>Login Page</h1>
      <button onClick={startLogin}>Login</button>
    </>
  );
};

const mapDispatchToProps = {
  startLogin: () => startLogin(),
};

export default connect(null, mapDispatchToProps)(Login);
