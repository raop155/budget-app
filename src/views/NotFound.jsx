import React from 'react';
import { Link } from 'react-router-dom';

export const NotFound = () => {
  return (
    <div>
      404 - <Link to='/'>Go home</Link>
    </div>
  );
};

export default NotFound;
