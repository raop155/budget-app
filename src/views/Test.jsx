import React from 'react';
import { BrowserRouter as Router, Switch, Route, useParams, useRouteMatch } from 'react-router-dom';
import Child1 from '../components/Child1';
import Child2 from '../components/Child2';

export const Test = (props) => {
  console.log(props);

  let { path, url } = useRouteMatch();

  console.log('path', path);
  console.log('url', url);

  let params = useParams();

  console.log(params);

  return (
    <Router>
      <h1>TEST</h1>
      <Switch>
        <Route path='/test/child1/:childId' component={Child1} />
        <Route path='/test/child2' component={Child2} />
        {/* <Route component={Child2} /> */}
      </Switch>
    </Router>
  );
};

export default Test;
