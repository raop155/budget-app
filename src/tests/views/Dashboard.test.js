import React from 'react';
import { shallow } from 'enzyme';
import { Dashboard } from '../../views/Dashboard';

describe('should render Dashboard view', () => {
  it('correctly', () => {
    const wrapper = shallow(<Dashboard />);
    expect(wrapper).toMatchSnapshot();
  });
});
