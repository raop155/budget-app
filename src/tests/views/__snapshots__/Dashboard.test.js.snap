// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Dashboard view correctly 1`] = `
<div>
  <Connect(ExpenseListFilters) />
  <Connect(ExpenseList) />
</div>
`;
